### PR TITLE
[MRG] fix BayesSearchCV on sklearn 0.19

### LINKED ISSUE #466: BayesSearchCV breaks with scikit-learn 0.19
BayesSearchCV relies on the privat ``_fit`` which will be removed in 0.19.

We should make it work with 0.18 and 0.19 and ideally use public interfaces only to be future-proof.

### PR DESCRIPTION
**Pull request by iaroslav-ai**
Fixes #466 

Move ._fit method inside the BayesSearchCV - maybe a temporary solution. Lets see if this breaks on Travis still.

### CODE REVIEW

**Comment by iaroslav-ai** [comment]
Seems to run fine with sklearn 0.18 and 0.19. Let me know what you think.

**Comment by iaroslav-ai** [comment]
#466

**Comment by betatim** [comment]
I'll take a look at this tomorrow.

**Comment by iaroslav-ai** [comment]
When you look at it later: I simply copied `_fit` method from 0.18.1 sklearn, as well as few properties which are not there anymore as it seems (but maybe I just missed something)

**Comment by betatim** [comment]
Ok. The purist in me would like to see if we can't make it work with the new multimetric support :)

File: skopt/searchcv.py [hunk 1]
```diff
@@ -398,6 +399,146 @@ def add_spaces(self, names, search_spaces):
         for space, name in zip(search_spaces, names):
             self.search_spaces_[name] = space
 
+    # copied for compatibility with 0.19 sklearn from 0.18 BaseSearchCV
+    @property
+    def best_score_(self):
+        sk_model_sel.check_is_fitted(self, 'cv_results_')
```
Comment (betatim) [inline skopt/searchcv.py]: we should import this as `from sklearn.utils.validation import check_is_fitted` instead of pulling it out of the model selection module.

File: skopt/searchcv.py [hunk 2]
```diff
@@ -398,6 +399,146 @@ def add_spaces(self, names, search_spaces):
         for space, name in zip(search_spaces, names):
             self.search_spaces_[name] = space
 
+    # copied for compatibility with 0.19 sklearn from 0.18 BaseSearchCV
+    @property
+    def best_score_(self):
```
Comment (betatim) [inline skopt/searchcv.py]: We don't use `best_score_` and `best_params_` for our code right? So I think we should not retrofit them, that is somehow between the user and sklearn I think. My motivation is to keep `BayesSearchCV` as similar as possible in interface to what ever version of sklearn the user has.

**Comment by betatim** [comment]
If we have to copy code could we copy `fit` from `BaseSearchCV` in 0.19? Then we would also support the new multi-metric functionality.

File: skopt/searchcv.py (hunk 2 above)
Comment (iaroslav-ai) [inline skopt/searchcv.py]: Actually these both are used in the code and examples, hence were breaking the circle ci

**Comment by iaroslav-ai** [comment]
Yeah there are some nicies like '**fit_params' in 0.19, which would be good to include now.

**Comment by iaroslav-ai** [comment]
I would delay including multi-metric support until https://github.com/scikit-learn/scikit-learn/issues/9499 is resolved. Just attempted to copy methods from `fit` of new `BaseSearchCV`, but it seems that either we need to copy and modify a lot of things, or we need to break compatibility with 0.18. 

Thus my suggestion would be to merge this (when fixed other issues) as temporary solution and update when corresponding base search is easier to extend.

File: skopt/searchcv.py (hunk 2 above)
Comment (betatim) [inline skopt/searchcv.py]: You are right we need to set them by hand if we copy the 0.18 code, in 0.19 they get set by `fit()`.

**Comment by betatim** [comment]
So merge and we create an issue to discuss how to do things in the future?

**Comment by iaroslav-ai** [comment]
Yes. Merged
